src/socket.ts

```diff
@@ -319,7 +328,13 @@ export class ConstellationSocket extends EventEmitter {
     }
 
     private rebroadcastEvent(name: string) {
-        this.socket.addEventListener(name, evt => this.emit(name, evt));
+        const socket = this.socket;
+        socket.addEventListener(name, evt => {
+            // Sanity check that the socket is current.
+            if (this.socket === socket) {
```

Comment:
Well, actually a reconnect state would still leave the .close() then .connect() issue and would neccesitate a .reconnect() method